### PR TITLE
QSP-2 Missing input checks

### PR DESCRIPTION
**Pull request by viraj124**
- Added missing checks in constructor for GoodGhosting.sol, GoodGhostingPolygon.sol and GoodGhostingPolygonWhitelisted
- updated unit tests to cover these scenario's

### CODE REVIEW

File: contracts/GoodGhosting.sol [hunk 1]
```diff
@@ -140,6 +140,12 @@ contract GoodGhosting is Ownable, Pausable {
         require(_earlyWithdrawalFee <= 10);
         require(_earlyWithdrawalFee > 0);
         require(_maxPlayersCount > 0, "_maxPlayersCount must be greater than zero");
+        require(address(_inboundCurrency) != address(0));
+        require(address(_lendingPoolAddressProvider) != address(0));
+        require(_segmentCount > 0);
+        require(_segmentLength > 0);
+        require(_segmentPayment > 0);
+        require(_dataProvider != address(0));
```
Comment (fodisi) [inline contracts/GoodGhosting.sol:148]: can we add the error message, just to stick to best practices? Thanks

**Review by fodisi — CHANGES_REQUESTED** [review]
Can we just add the error message to the reverts, and then check the error messages in our tests as well?

Just to comply with solidity best practices.
Thanks

File: contracts/GoodGhosting.sol (hunk 1 above)
Comment (viraj124) [reply contracts/GoodGhosting.sol:148]: sure

Comment (fodisi) [reply contracts/GoodGhosting.sol:148]: thanks :)

**Review by fodisi — APPROVED** [review]

**Pull request merged**